### PR TITLE
Experimental - Situation invoice - Use progress mode (GIFF)

### PR DESCRIPTION
**Pull request by aspangaro**
New progressive mode for situation invoice. Goal is to replace cumulative mode.

Constant INVOICE_USE_SITUATION must be defined to 2

### CODE REVIEW

File: htdocs/compta/facture/class/facture.class.php [hunk 1]
```diff
@@ -6721,6 +6735,291 @@ public function update_total()
...
+		} elseif ($this->element == 'facture_fourn' || $this->element == 'supplier_invoice' || $this->element == 'invoice_supplier' || $this->element == 'invoice_supplier_rec') {
+			$MODULE = "MODULE_DISALLOW_UPDATE_PRICE_SUPPLIER_INVOICE";
+		} elseif ($this->element == 'order_supplier' || $this->element == 'supplier_order') {
+			$MODULE = "MODULE_DISALLOW_UPDATE_PRICE_SUPPLIER_ORDER";
+		} elseif ($this->element == 'supplier_proposal') {
+			$MODULE = "MODULE_DISALLOW_UPDATE_PRICE_SUPPLIER_PROPOSAL";
+		}
+
+		if (!empty($MODULE)) {
+			if (!empty($conf->global->$MODULE)) {
```
Comment (eldy) [inline htdocs/compta/facture/class/facture.class.php:6779]: Not compatible with recent php version.
Can you replace with
if (getDolGlobalString($MODULE))

Ps: as this is addition of a new function, why not having submitting it into a separated PR. Splitting large PR into several ones increase dramatically speed and ease of integration.

**Review by eldy — CHANGES_REQUESTED** [review]

File: htdocs/compta/facture/class/facture.class.php (hunk 1 above)
Comment (aspangaro) [reply htdocs/compta/facture/class/facture.class.php]: Thanks @eldy. Fix pushed

File: htdocs/langs/en_US/bills.lang [hunk 2]
```diff
@@ -238,6 +238,7 @@ RetainedwarrantyOnlyForSituationFinal=On situation invoices the global "retained
 ToPayOn=To pay on %s
 toPayOn=to pay on %s
 RetainedWarranty=Retained Warranty
+RetainedWarrantyShort=Ret. warranty
```
Comment (eldy) [inline htdocs/langs/en_US/bills.lang:241]: Duplicate entry

Comment (aspangaro) [reply htdocs/langs/en_US/bills.lang]: You're right.

File: htdocs/compta/facture/card.php [hunk 3]
```diff
@@ -2849,7 +2925,7 @@
 		if (GETPOST('all_progress') != "") {
 			$all_progress = GETPOSTINT('all_progress');
 			foreach ($object->lines as $line) {
-				$percent = $line->get_prev_progress($object->id);
+				$percent = $line->get_allprev_progress($object->id);
```
[the remark this answers is not in the document]
Comment (aspangaro) [reply htdocs/compta/facture/card.php]: Fixed ! Thanks

**Pull request closed without merging**